network/p2p/gossip/handler.go

```diff
@@ -124,3 +135,31 @@ func (h Handler[_]) AppGossip(_ context.Context, nodeID ids.NodeID, gossipBytes
 		)
 	}
 }
+
+func computeBloomFilterHitPercentage(hits uint64, misses uint64, log logging.Logger) (uint64, bool) {
+	total, err := safemath.Add(hits, misses)
+	if err != nil {
+		log.Warn("failed to calculate total hits and misses",
+			zap.Uint64("hits", hits),
+			zap.Uint64("misses", misses),
+			zap.Error(err),
+		)
+		return 0, false
+	}
+
+	hitsOneHundred, err := safemath.Mul(hits, 100)
+	if err != nil {
+		log.Warn("failed to calculate hit ratio",
+			zap.Uint64("hits", hits),
+			zap.Uint64("misses", misses),
+			zap.Error(err),
+		)
+		return 0, false
+	}
```

Comment:
I wonder if we can get rid of these overflow checks... since the bloom filter is `[]byte` (which can be at most of size max int) could we just emit a metric on both the hits + length of the filter? We could figure out the hit/miss % in dashboards with metric math.

Reply:
> I wonder if we can get rid of these overflow checks... since the bloom filter is []byte (which can be at most of size max int) could we just emit a metric on both the hits + length of the filter? We could figure out the hit/miss % in dashboards with metric math.

I am not sure what you mean by the length of the filter. The bloom filter is constant size and its length is always the same, but yet independent of the elements that it can "contain". 

I guess you meant the size of the mempool? Since the maximum number of hits we can get is the size of the mempool, not the bloom filter. Sure I can do that, that's a good point.

Reply:
Unfortunately we don't have a Size() on the set interface so I will try to code it without the safemath.

Reply:
[Changed](https://github.com/ava-labs/avalanchego/pull/4085/commits/b21c5a553095c4a7feb11523272b1dc8f3ca336f#diff-64f640fd349509626d5061afbee4d706338a2f04f12a0bca5cf2a96447d614d5) the code to not use safemath.